frame/executive/src/lib.rs

```diff
@@ -377,7 +398,29 @@ where
 		}
 	}
 
+	#[cfg(not(feature = "background-sig-check"))]
+	/// Actually execute all transitions for `block`.
+	pub fn execute_block(block: Block) {
+		sp_io::init_tracing();
+		sp_tracing::within_span! {
+			sp_tracing::info_span!("execute_block", ?block);
+
+			Self::initialize_block(block.header());
+
+			// any initial checks
+			Self::initial_checks(&block);
+
+			// execute extrinsics
+			let (header, extrinsics) = block.deconstruct();
+			Self::execute_extrinsics_with_book_keeping(extrinsics, *header.number());
+
+			// any final checks
+			Self::final_checks(&header);
+		}
+	}
+
 	/// Execute given extrinsics and take care of post-extrinsics book-keeping.
+	#[cfg(not(feature = "background-sig-check"))]
 	fn execute_extrinsics_with_book_keeping(
```

Comment:
This can also be merged with your method when we make a little bit more generic.

Reply:
Done